OWID/OWID_Visualize_GDP_per_capita_through_the_years.ipynb

```diff
@@ -0,0 +1,271 @@
+   "metadata": {},
+   "source": [
+    "### Get Data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "aad6eb74",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "contents_df = pd.read_excel(url, sheet_name = \"Data\")\n",
+    "new_column = []\n",
+    "for i in range(len(contents_df)):\n",
+    "    if contents_df.iloc[i,4] == np.nan:\n",
+    "        new_column.append(0)\n",
+    "    else:\n",
+    "        new_column.append(contents_df.iloc[i,4]/contents_df.iloc[i,6])\n",
+    "new_dict = {\"Country\": contents_df[\"country\"],\n",
+    "           \"Year\": contents_df[\"year\"],\n",
+    "           \"gdp\": new_column}\n",
+    "new_df = pd.DataFrame(data=new_dict)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "560e3784",
+   "metadata": {},
+   "source": [
+    "### Create Dash App"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c2629798",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "app = Dash(title = \"GDP per capita\")\n",
+    "p = \"gdp\"\n",
+    "app.layout = html.Div(\n",
+    "    [\n",
+    "        html.H4(\"GDP per capita from 1950 to 2019\"),\n",
+    "        html.P(\"Select countries\"),\n",
+    "        dcc.Dropdown(\n",
+    "            id=\"countries\",\n",
+    "            options=new_df.Country.unique(),\n",
+    "            value=[\"United States\"],\n",
+    "            multi = True\n",
+    "        ),\n",
+    "        dcc.RangeSlider(id='slider', min=1950, max=2019, value=[1950, 2019],\n",
+    "               marks={x: str(x) for x in [1950, 1965, 1980, 1995, 2010, 2019]}),\n",
+    "        dcc.Graph(id=p, figure={}, style={'display': 'none'})\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "@callback(\n",
+    "    Output(p, 'figure'),\n",
+    "    Output(p, 'style'),\n",
+    "    Input('countries', 'value'),\n",
+    "    Input('slider', 'value')\n",
+    ")\n",
+    "\n",
+    "def update_graph(country, year):\n",
+    "    dff = (new_df[\"Country\"].isin(country)) & (new_df[\"Year\"] <= year[1]) & (new_df[\"Year\"] >= year[0])\n",
+    "    figures = px.line(new_df[dff], x='Year', y=p, color = \"Country\", markers = True).update_layout(\n",
+    "                                plot_bgcolor='rgba(0, 0, 0, 0)', height= 600\n",
+    "                            )\n",
+    "    styles = {'display': 'block'}   \n",
+    "    return figures, styles\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d43335ed",
+   "metadata": {},
+   "source": [
+    "## Output"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4f1ab927",
+   "metadata": {},
+   "source": [
+    "### Generate URL and show logs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "75639a64",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if __name__ == \"__main__\":\n",
+    "    app.run_server(proxy=f\"http://127.0.0.1:{DASH_PORT}::https://app.naas.ai\")"
+   ]
+  }
```

Comment:
Also for the output, I get a black frame and nothing is visible. Would you mind creating a small loom recording of the template working? That would be helpful in reviewing the PR.

<img width="1377" alt="image" src="https://github.com/jupyter-naas/awesome-notebooks/assets/81156510/3d755c10-6ada-4766-896d-f70bc6bc251d">

Reply:
This I actually don't know what is the issue here, since when I ran the notebook, a link will pop up and you can see a plot.

Reply:
@makingsurgeon Sorry for the delay in response. Did you use the Naas platform for testing? If so can you share what browser did you use? That would be helpful in testing on my end.

Reply:
@srini047 I ran on my local machine since I couldn't configure Naas to work. But I copied the code related to Naas from this: https://github.com/jupyter-naas/awesome-notebooks/blob/master/OWID/OWID_Visualize_Population_of_Different_Age_Groups.ipynb and it seems that you are able to run this without issue on Naas.

Reply:
Okay, thanks for confirming. Yes, it did run without any issues but the output is not displayed.

It's not that difficult to configure Naas. You can head to [Naas cloud](https://app.naas.ai) create an account if it doesn't exist or log in and then run the same notebook.

You can just record the screen to showcase its working because it will be helpful for others using the same notebook.

If you are stuck anywhere please feel free to tag me😊

Reply:
I added Florent to the list of reviewers, since I know the problem probably lies in the code that you screeenshoted. I experienced the same problem when I tried on Naas, but the weird thing is that that code can be found in many Dash notebooks, so maybe Florent will have a better idea.

Reply:
@makingsurgeon  @srini047, i just checked the template and found where was the issue. The app was initialize a second time in the create app cell. I removed the code `app = dash.Dash(title = "GDP per capita")` and it is now working well!
Thank you for your contribution @makingsurgeon  and your review @srini047 !